src/compiler/parser.ts

```diff
@@ -5302,6 +5302,10 @@ namespace ts {
             node.dotDotDotToken = parseOptionalToken(SyntaxKind.DotDotDotToken);
             const tokenIsIdentifier = isIdentifier();
             const propertyName = parsePropertyName();
+            if (token() === SyntaxKind.QuestionToken) {
+                parseErrorAtCurrentToken(Diagnostics.Property_in_destructuring_not_be_marked_optional);

```

Comment:
we do not usually error in the parser. this impacts our ability to do incremental parsing correctly. instead we will parse it, then add a grammerCheck in the checker for it.